assignment_2.java

```diff
@@ -0,0 +1,144 @@
+/***
+ *When the program is run, the user will first enter a string. After that, the program should
+ask the user which operation they want to execute on the string. The available operations are
+listed below: Append, CountWords, Replace, isPalindrome, Splice, Split,
+MaxRepeatingCharacter, Sort, Shift, and Reverse. Each method should perform its
+respective operation based on the user's input.
+Owner : Pratul_Agarwal
+Date of creation : 04/09/2024
+ */
+package assignment_java_programming;
+
+import java.util.Scanner;
+
+public class assignment_2 {
+
+    public static void main(String[] args) {
+    	constant constant = new constant();
+        Scanner scanner = new Scanner(System.in);
+        String tryagain = "y";
+        
+        while (tryagain.equalsIgnoreCase("y")) {
+        System.out.println(constant.ENTER_STRING6);
+        String input = scanner.nextLine();
+
+        // Present operations to the user
+        System.out.println("Which operation you want to execute on the string.");
+        System.out.println("1. Append");
+        System.out.println("3. Replace");
+        System.out.println("4. isPalindrome");
+        System.out.println("5. Splice");
+        System.out.println("7. MaxRepeatingCharacter");
+        System.out.println("8. Sort");
+        System.out.println("9. Shift");
+        System.out.println("10. Reverse");
+
+        int choice = scanner.nextInt();
+        scanner.nextLine();  
+
+        switch (choice) {
+            case 1:
+                System.out.println("Enter string to append:");
+                String appendStr = scanner.nextLine();
+                System.out.println("Result: " +Append_String(input, appendStr));
+                break;
+            case 3:
+                System.out.println("Enter target substring to replace:");
+                String target = scanner.nextLine();
+                System.out.println("Enter replacement substring:");
+                String replacement = scanner.nextLine();
+                System.out.println("Result: " + replace(input, target, replacement));

```

Comment:
Method name Append_String should be appendString (camelCase). Consistency in method naming is crucial.